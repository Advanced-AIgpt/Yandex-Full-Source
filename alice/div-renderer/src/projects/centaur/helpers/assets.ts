export function getS3Asset(path: string) {
    return `https://dialogs.s3.yandex.net/smart_displays/${path}`;
}

export function getStaticS3Asset(path: string) {
    return `https://yastatic.net/s3/dialogs/smart_displays/${path}`;
}
