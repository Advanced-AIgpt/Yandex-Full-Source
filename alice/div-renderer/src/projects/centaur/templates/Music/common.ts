import {
    ContainerBlock,
    Div,
    FixedSize,
    IDivAction,
    ImageBackground,
    ImageBlock, MatchParentSize,
    SolidBackground,
    TextBlock,
} from 'divcard2';
import { colorBlackOpacity70, offsetFromEdgeOfScreen } from '../../style/constants';
import { compact } from 'lodash';
import EmptyDiv from '../../components/EmptyDiv';
import { text28m } from '../../style/Text/Text';
import { directivesAction } from '../../../../common/actions';
import { centaurMusicPlay } from '../../../../common/actions/server/musicActions';
import MusicPlaying from './MusicPlaying/MusicPlaying';
import { IRequestState } from '../../../../common/types/common';
import LikeButton from './components/LikeButton';

interface IGetPlaylistItemProps {
    title: string;
    subtitle?: string;
    objectId: string;
    objectType: string;
    startFromTrackId?: string;
    isCurrent?: boolean;
    imageUrl: string;
    imageBackground?: string;
    requestState: IRequestState;
    isLiked?: boolean;
}

function CoverWithBackground(imageUrl: string, background: string, action: IDivAction, isCurrent: boolean, requestState: IRequestState) {
    return new ContainerBlock({
        width: new FixedSize({ value: 80 }),
        height: new FixedSize({ value: 80 }),
        alignment_vertical: 'top',
        actions: [action],
        margins: {
            right: 24,
        },
        border: {
            corner_radius: 40,
        },
        content_alignment_horizontal: 'center',
        content_alignment_vertical: 'center',
        background: compact([
            new SolidBackground({
                color: background,
            }),
        ]),
        orientation: 'overlap',
        items: compact([
            new ImageBlock({
                image_url: imageUrl,
                width: new FixedSize({ value: 48 }),
                height: new FixedSize({ value: 48 }),
            }),
            isCurrent && new EmptyDiv({
                width: new FixedSize({ value: 80 }),
                height: new FixedSize({ value: 80 }),
                background: [
                    new SolidBackground({
                        color: colorBlackOpacity70,
                    }),
                ],
            }),
            isCurrent && MusicPlaying(requestState),
        ]),
    });
}

function CoverWithoutBackground(imageUrl: string, action: IDivAction, isCurrent: boolean, requestState: IRequestState) {
    return new ContainerBlock({
        width: new FixedSize({ value: 80 }),
        height: new FixedSize({ value: 80 }),
        alignment_vertical: 'top',
        content_alignment_horizontal: 'center',
        content_alignment_vertical: 'center',
        actions: [action],
        margins: {
            right: 24,
        },
        border: {
            corner_radius: 20,
        },
        background: compact([
            new ImageBackground({
                image_url: imageUrl,
            }),
            isCurrent && new SolidBackground({
                color: colorBlackOpacity70,
            }),
        ]),
        items: [
            isCurrent ? MusicPlaying(requestState) : new EmptyDiv(),
        ],
    });
}

export function getPlaylistItem({
    isCurrent = false,
    objectType,
    subtitle,
    objectId,
    title,
    startFromTrackId,
    imageUrl,
    imageBackground,
    requestState,
    isLiked,
}: IGetPlaylistItemProps): {
    options: Partial<ConstructorParameters<typeof ContainerBlock>[0]>;
    items: Div[];
} {
    const openAction = {
        log_id: 'play_music_click',
        url: directivesAction(centaurMusicPlay({
            disableNlg: true,
            objectType,
            objectId,
            startFromTrackId,
        })),
    };

    return {
        options: {
            orientation: 'horizontal',
            paddings: {
                top: 28,
                left: offsetFromEdgeOfScreen,
                bottom: 28,
                right: offsetFromEdgeOfScreen,
            },
            content_alignment_vertical: 'center',
        },
        items: compact([
            imageBackground ?
                CoverWithBackground(imageUrl, imageBackground, openAction, isCurrent, requestState) :
                CoverWithoutBackground(imageUrl, openAction, isCurrent, requestState),
            new ContainerBlock({
                width: new MatchParentSize({ weight: 1 }),
                actions: [openAction],
                items: compact([
                    new TextBlock({
                        ...text28m,
                        text: title,
                    }),
                    subtitle && new TextBlock({
                        ...text28m,
                        text: subtitle,
                        alpha: 0.5,
                    }),
                ]),
            }),
            typeof isLiked !== 'undefined' && LikeButton({ isLiked, trackId: startFromTrackId || '' }),
        ]),
    };
}
