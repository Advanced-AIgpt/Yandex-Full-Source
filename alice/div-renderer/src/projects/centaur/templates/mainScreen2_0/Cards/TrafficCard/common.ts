import { ITrafficForecast } from './types';

export function sortTimeOfDay(originalArray: Readonly<ITrafficForecast[]>): ITrafficForecast[] {
    const forecastsFirst = [];
    const forecastsLast = [];

    for (const forecastElement of originalArray) {
        if (forecastElement.time === null || typeof forecastElement.time === 'undefined') {
            continue;
        }

        if (forecastsFirst.length === 0) {
            forecastsFirst.push(forecastElement);
        } else if (forecastsLast.length > 0) {
            forecastsLast.push(forecastElement);
        } else if (forecastElement.time - (forecastsFirst[forecastsFirst.length - 1].time as number) === 1) {
            forecastsFirst.push(forecastElement);
        } else {
            forecastsLast.push(forecastElement);
        }
    }

    return [...forecastsLast, ...forecastsFirst];
}
