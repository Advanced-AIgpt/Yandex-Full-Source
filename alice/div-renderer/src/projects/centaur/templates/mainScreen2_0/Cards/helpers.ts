export function getCardMainScreenId({ rowIndex, colIndex }: { rowIndex?: number; colIndex?: number }) {
    return `main_screen_card_${rowIndex || 0}_${colIndex || 0}`;
}
