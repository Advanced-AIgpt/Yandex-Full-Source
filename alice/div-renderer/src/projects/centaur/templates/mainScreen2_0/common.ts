import { IDivAction } from 'divcard2';
import { EnumLayer } from '../../actions/client';
import { CrutchLoader } from '../../components/Loader/Loader';

interface IPutLoaderToActionOptions {
    action?: string | null,
    layer: EnumLayer,
    actionLogId?: string,
}

export const putLoaderToAction = ({
    action,
    layer,
    actionLogId = 'action_with_loader',
}: IPutLoaderToActionOptions): IDivAction[] | undefined => {
    if (!action) {
        return undefined;
    }
    return [
        CrutchLoader(layer),
        {
            log_id: actionLogId,
            url: action,
        },
    ];
};
