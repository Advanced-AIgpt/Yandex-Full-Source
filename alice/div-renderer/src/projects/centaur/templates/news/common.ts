import {
    ContainerBlock,
    DivBackground,
    FixedSize, IDivAction,
    ImageBackground,
    ImageBlock,
    MatchParentSize,
    SolidBackground,
    TextBlock,
    WrapContentSize,
} from 'divcard2';
import { compact } from 'lodash';
import { utcToZonedTime } from 'date-fns-tz';
import { differenceInCalendarDays, isSameDay } from 'date-fns';
import { Long } from 'protobufjs';
import { NAlice } from '../../../../protos';
import { formatDate } from '../../helpers/helpers';
import {
    colorDarkGrey1Opacity70,
    colorReadButton,
    colorWhite,
    colorWhiteOpacity60,
    offsetFromEdgeOfScreen,
} from '../../style/constants';
import { text40r, title32m, title60m } from '../../style/Text/Text';
import { getImageByTopic } from '../../helpers/newsHelper';
import { Avatar } from '../../../../common/helpers/avatar';

export const patchToOrig = (avatarsUrl: string) => avatarsUrl.replace(/\/[^/]+$/, '/orig');

export type NewsCardProps = {
    item: NAlice.NData.ITNewsItem;
    topic: string;
    bottomGap: boolean;
    tz: string;
};

export function newsCommonCard(
    { item: { Text, Image, Agency, Logo, PubDate, ExtendedNews }, topic, bottomGap, tz }: NewsCardProps,
    openExtendedNewsDivAction: IDivAction) {
    return new ContainerBlock({
        width: new MatchParentSize(),
        height: new MatchParentSize(),
        background: generateBackground(topic, Image),
        content_alignment_horizontal: 'left',
        content_alignment_vertical: 'bottom',
        paddings: {
            left: offsetFromEdgeOfScreen,
            bottom: bottomGap ? 104 : offsetFromEdgeOfScreen,
            right: offsetFromEdgeOfScreen,
        },
        items: [
            new ContainerBlock({
                width: new WrapContentSize(),
                height: new MatchParentSize({ weight: 1 }),
                content_alignment_vertical: 'bottom',
                orientation: 'horizontal',
                margins: {
                    bottom: 20,
                },
                items: newsSource(Agency, Logo, PubDate, tz),
            }),
            new ContainerBlock({
                width: new MatchParentSize(),
                height: new WrapContentSize(),
                content_alignment_vertical: 'bottom',
                orientation: 'horizontal',
                items: compact([
                    new TextBlock({
                        ...title60m,
                        width: new MatchParentSize({ weight: 1 }),
                        height: new WrapContentSize(),
                        max_lines: 5,
                        truncate: 'end',
                        text: Text ?? ' ',
                    }),
                    ExtendedNews?.length && new TextBlock({
                        ...text40r,
                        width: new WrapContentSize(),
                        height: new WrapContentSize(),
                        background: [new SolidBackground({
                            color: colorReadButton,
                        })],
                        border: {
                            corner_radius: 24,
                        },
                        paddings: {
                            top: 22,
                            right: 33,
                            bottom: 22,
                            left: 33,
                        },
                        margins: {
                            left: 48,
                        },
                        alignment_horizontal: 'right',
                        text: 'Читать',
                        action: openExtendedNewsDivAction,
                    }),
                ]),
            }),
        ],
    });
}

export const newsSource = (
    Agency: (string | null | undefined),
    Logo: (string | null | undefined),
    PubDate: (number | Long | null | undefined),
    tz: string,
) =>
    compact([
        Logo && new ContainerBlock({
            background: [
                new SolidBackground({
                    color: colorWhite,
                }),
            ],
            content_alignment_horizontal: 'center',
            content_alignment_vertical: 'center',
            border: {
                corner_radius: 12,
            },
            width: new FixedSize({ value: 40 }),
            height: new FixedSize({ value: 40 }),
            margins: {
                right: 20,
            },
            items: [
                new ImageBlock({
                    width: new FixedSize({ value: 28 }),
                    height: new FixedSize({ value: 28 }),
                    image_url: Logo ?? '',
                }),
            ],
        }),
        new TextBlock({
            ...title32m,
            text_color: colorWhiteOpacity60,
            width: new WrapContentSize(),
            height: new WrapContentSize(),
            text_alignment_vertical: 'center',
            text: (Agency ? Agency + ' • ' : '') + (Number(PubDate) ? formDate(Number(PubDate) * 1000 || 0, tz) : ''),
        }),
    ]);

export const generateBackground = (
    Topic: string | null | undefined,
    NewsImage: NAlice.NData.TNewsItem.ITImage | null | undefined,
    color = colorDarkGrey1Opacity70,
) => {
    const result: DivBackground[] = [];
    if (NewsImage && NewsImage.Src) {
        const avatar = Avatar.fromUrl(NewsImage.Src);

        result.push(
            new ImageBackground({
                image_url: avatar ? avatar.setTypeName('orig').toString() : NewsImage.Src,
                scale: 'fill',
                preload_required: 1,
            }),
        );
    } else {
        result.push(
            new ImageBackground({
                image_url: getImageByTopic(Topic),
                preload_required: 1,
            }),
        );
    }

    result.push(new SolidBackground({ color }));
    return result;
};

export const formDate = (date: number, tz: string) => {
    const d = utcToZonedTime(date, tz);
    const now = utcToZonedTime(Date.now(), tz);
    if (isSameDay(d, now)) {
        return formatDate(d, 'HH:mm');
    }
    if (differenceInCalendarDays(now, d) === 1) {
        return `вчера в ${formatDate(d, 'HH:mm')}`;
    }
    return formatDate(d, 'd MMMM в HH:mm');
};
