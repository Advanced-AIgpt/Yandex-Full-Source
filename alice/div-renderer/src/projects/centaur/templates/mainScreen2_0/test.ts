import MainScreen2_0 from './index';
import { AnonymizeDataForSnapshot } from '../../../../common/helpers/dev';
import { NAlice } from '../../../../protos';
import { getS3Asset } from '../../helpers/assets';
import { MMRequest } from '../../../../common/helpers/MMRequest';
import { createRequestState } from '../../../../registries/common';

const TColumn = NAlice.NData.TCentaurMainScreenMyScreenData.TColumn;
const TCardData = NAlice.NData.TCentaurMainScreenMyScreenData.TColumn.TCardData;

jest.mock('../../../../common/logger');

const mmRequest = new MMRequest({}, {}, {});

describe('Main screen', () => {
    it('check snapshot without columns', () => {
        expect(AnonymizeDataForSnapshot(MainScreen2_0({
            Columns: [],
        }, mmRequest, createRequestState()))).toMatchSnapshot();
    });

    it('check snapshot without rows', () => {
        expect(AnonymizeDataForSnapshot(MainScreen2_0({
            Columns: [
                new TColumn({
                    Cards: [],
                }),
                new TColumn({
                    Cards: [],
                }),
                new TColumn({
                    Cards: [],
                }),
            ],
        }, mmRequest, createRequestState()))).toMatchSnapshot();
    });

    it('check snapshot with example data', () => {
        expect(AnonymizeDataForSnapshot(MainScreen2_0({
            Columns: [
                new TColumn({
                    Cards: [
                        new TCardData({
                            MusicCardData: {
                                Name: 'Плейлист дня',
                                Modified: {
                                    value: (new Date()).toString(),
                                },
                                Cover: '1',
                                Color: '#31AD54',
                            },
                        }),
                    ],
                }),
                new TColumn({
                    Cards: [
                        new TCardData({
                            InfoCardData: {
                                Title: 'test',
                                Description: 'Алиса, включи шум морской волны',
                                ImageBackground: getS3Asset('discovery/motivation.png'),
                                Color: '#3478C2',
                            },
                        }),
                        new TCardData({
                            WeatherCardData: {
                                Temperature: -3,
                                City: 'Москва',
                                Image: '1',
                                Comment: 'Пасмурно.\nОщущается как -5°',
                            },
                        }),
                    ],
                }),
                new TColumn({
                    Cards: [
                        new TCardData({
                            YouTubeCardData: {},
                        }),
                        new TCardData({
                            NewsCardData: {
                                Title: 'Медуза',
                                Content: 'Tesla начала устанавливать терминалы Starlink на зарядных станциях Supercharger',
                            },
                        }),
                    ],
                }),
            ],
        }, mmRequest, createRequestState()))).toMatchSnapshot();
    });
});
