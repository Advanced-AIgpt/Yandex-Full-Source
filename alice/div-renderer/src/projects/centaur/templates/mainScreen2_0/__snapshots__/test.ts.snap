// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main screen check snapshot with example data 1`] = `
Object {
  "card": Object {
    "log_id": "mainScreen",
    "states": Array [
      Object {
        "div": Object {
          "background": Array [
            Object {
              "color": "#121316",
              "type": "solid",
            },
          ],
          "height": Object {
            "type": "match_parent",
          },
          "items": Array [
            Object {
              "height": Object {
                "type": "match_parent",
              },
              "items": Array [
                Object {
                  "height": Object {
                    "type": "match_parent",
                  },
                  "items": Array [
                    Object {
                      "background": Array [
                        Object {
                          "color": "#31AD54",
                          "type": "solid",
                        },
                      ],
                      "border": Object {
                        "corner_radius": 28,
                      },
                      "height": Object {
                        "type": "match_parent",
                        "weight": 1,
                      },
                      "id": "main_screen_card_0_0",
                      "items": Array [
                        Object {
                          "font_size": 32,
                          "font_weight": "medium",
                          "line_height": 40,
                          "text": "Рекомендуем",
                          "text_color": "#80ffffff",
                          "type": "text",
                        },
                        Object {
                          "delimiter_style": Object {
                            "color": "#0fff",
                          },
                          "height": Object {
                            "type": "match_parent",
                          },
                          "type": "separator",
                        },
                        Object {
                          "alignment_horizontal": "center",
                          "border": Object {
                            "corner_radius": 18,
                          },
                          "height": Object {
                            "type": "fixed",
                            "value": 220,
                          },
                          "image_url": "1",
                          "type": "image",
                          "width": Object {
                            "type": "fixed",
                            "value": 220,
                          },
                        },
                        Object {
                          "font_size": 32,
                          "font_weight": "medium",
                          "line_height": 40,
                          "margins": Object {
                            "top": 24,
                          },
                          "text": "Плейлист дня",
                          "text_alignment_horizontal": "center",
                          "text_color": "#fff",
                          "type": "text",
                        },
                        Object {
                          "font_size": 32,
                          "font_weight": "medium",
                          "line_height": 40,
                          "margins": Object {
                            "top": 4,
                          },
                          "text": "Обновлен сегодня",
                          "text_alignment_horizontal": "center",
                          "text_color": "#80ffffff",
                          "type": "text",
                        },
                        Object {
                          "delimiter_style": Object {
                            "color": "#0fff",
                          },
                          "height": Object {
                            "type": "match_parent",
                          },
                          "type": "separator",
                        },
                        Object {
                          "alignment_horizontal": "center",
                          "height": Object {
                            "type": "fixed",
                            "value": 79,
                          },
                          "image_url": "https://dialogs.s3.yandex.net/smart_displays/player/play.png?v=1",
                          "margins": Object {
                            "bottom": 8,
                          },
                          "type": "image",
                          "width": Object {
                            "type": "fixed",
                            "value": 79,
                          },
                        },
                      ],
                      "paddings": Object {
                        "bottom": 24,
                        "left": 24,
                        "right": 24,
                        "top": 24,
                      },
                      "type": "container",
                      "width": Object {
                        "type": "match_parent",
                        "weight": 1,
                      },
                    },
                  ],
                  "margins": Object {
                    "top": 0,
                  },
                  "type": "container",
                  "width": Object {
                    "type": "match_parent",
                  },
                },
              ],
              "margins": Object {
                "left": 0,
              },
              "type": "container",
              "width": Object {
                "type": "match_parent",
              },
            },
            Object {
              "height": Object {
                "type": "match_parent",
              },
              "items": Array [
                Object {
                  "height": Object {
                    "type": "match_parent",
                  },
                  "items": Array [
                    Object {
                      "actions": Array [
                        Object {
                          "log_id": "loader_click",
                          "url": "dialog://?directives=%5B%7B%22type%22%3A%22client_action%22%2C%22name%22%3A%22show_view%22%2C%22payload%22%3A%7B%22layer%22%3A%7B%22content%22%3A%7B%7D%7D%2C%22div2_card%22%3A%7B%22body%22%3A%7B%22templates%22%3A%7B%7D%2C%22card%22%3A%7B%22log_id%22%3A%22loader%22%2C%22states%22%3A%5B%7B%22state_id%22%3A0%2C%22div%22%3A%7B%22type%22%3A%22container%22%2C%22height%22%3A%7B%22type%22%3A%22match_parent%22%2C%22weight%22%3A1%7D%2C%22items%22%3A%5B%7B%22type%22%3A%22gif%22%2C%22alignment_horizontal%22%3A%22center%22%2C%22alignment_vertical%22%3A%22center%22%2C%22gif_url%22%3A%22https%3A%2F%2Fdialogs.s3.yandex.net%2Fsmart_displays%2Floaders%2Fspinner.gif%22%2C%22height%22%3A%7B%22type%22%3A%22fixed%22%2C%22value%22%3A250%7D%2C%22width%22%3A%7B%22type%22%3A%22fixed%22%2C%22value%22%3A250%7D%7D%2C%7B%22type%22%3A%22container%22%2C%22alignment_horizontal%22%3A%22right%22%2C%22alignment_vertical%22%3A%22top%22%2C%22items%22%3A%5B%7B%22type%22%3A%22image%22%2C%22actions%22%3A%5B%7B%22log_id%22%3A%22close_fullscreen%22%2C%22url%22%3A%22centaur%3A%2F%2Flocal_command%3Flocal_commands%3D%5B%257B%2522command%2522%253A%2522close_layer%2522%252C%2522layer%2522%253A%2522CONTENT%2522%252C%2522stop_conversation%2522%253Atrue%257D%5D%22%7D%5D%2C%22alignment_horizontal%22%3A%22right%22%2C%22background%22%3A%5B%7B%22type%22%3A%22solid%22%2C%22color%22%3A%22%231affffff%22%7D%5D%2C%22border%22%3A%7B%22corner_radius%22%3A20%7D%2C%22height%22%3A%7B%22type%22%3A%22fixed%22%2C%22value%22%3A88%7D%2C%22image_url%22%3A%22https%3A%2F%2Fyastatic.net%2Fs3%2Fdialogs%2Fsmart_displays%2Ficons%2Fclose_wt_border.png%22%2C%22margins%22%3A%7B%22top%22%3A36%2C%22right%22%3A36%7D%2C%22paddings%22%3A%7B%22top%22%3A22%2C%22right%22%3A22%2C%22left%22%3A22%2C%22bottom%22%3A22%7D%2C%22preload_required%22%3A1%2C%22scale%22%3A%22fill%22%2C%22width%22%3A%7B%22type%22%3A%22fixed%22%2C%22value%22%3A88%7D%7D%5D%7D%5D%2C%22orientation%22%3A%22overlap%22%2C%22width%22%3A%7B%22type%22%3A%22match_parent%22%2C%22weight%22%3A1%7D%7D%7D%5D%2C%22variable_triggers%22%3A%5B%5D%2C%22variables%22%3A%5B%5D%7D%7D%7D%2C%22do_not_show_close_button%22%3Anull%2C%22inactivity_timeout%22%3A%22Infinity%22%7D%7D%5D",
                        },
                        Object {
                          "log_id": "discovery.mainscreen.execute",
                          "url": "dialog://text_command?query=%D0%A0%D0%B0%D1%81%D1%81%D0%BA%D0%B0%D0%B6%D0%B8%20%D0%BD%D0%BE%D0%B2%D0%BE%D1%81%D1%82%D0%B8%20%D1%82%D0%B5%D1%85%D0%BD%D0%BE%D0%BB%D0%BE%D0%B3%D0%B8%D0%B9",
                        },
                      ],
                      "background": Array [
                        Object {
                          "content_alignment_horizontal": "right",
                          "image_url": "https://dialogs.s3.yandex.net/smart_displays/discovery/news.png",
                          "scale": "fill",
                          "type": "image",
                        },
                      ],
                      "border": Object {
                        "corner_radius": 28,
                      },
                      "height": Object {
                        "type": "match_parent",
                        "weight": 1,
                      },
                      "id": "main_screen_card_0_1",
                      "items": Array [
                        Object {
                          "items": Array [
                            Object {
                              "font_size": 32,
                              "font_weight": "medium",
                              "line_height": 40,
                              "text": "Новости",
                              "text_color": "#80ffffff",
                              "type": "text",
                            },
                          ],
                          "orientation": "horizontal",
                          "type": "container",
                        },
                        Object {
                          "delimiter_style": Object {
                            "color": "#0fff",
                          },
                          "height": Object {
                            "type": "match_parent",
                          },
                          "type": "separator",
                        },
                        Object {
                          "font_size": 28,
                          "font_weight": "medium",
                          "line_height": 36,
                          "text": "Расскажи новости технологий",
                          "text_color": "#e6ffffff",
                          "type": "text",
                        },
                      ],
                      "paddings": Object {
                        "bottom": 24,
                        "left": 24,
                        "right": 24,
                        "top": 24,
                      },
                      "type": "container",
                      "width": Object {
                        "type": "match_parent",
                        "weight": 1,
                      },
                    },
                  ],
                  "margins": Object {
                    "top": 0,
                  },
                  "type": "container",
                  "width": Object {
                    "type": "match_parent",
                  },
                },
                Object {
                  "height": Object {
                    "type": "match_parent",
                  },
                  "items": Array [
                    Object {
                      "actions": Array [],
                      "background": Array [
                        Object {
                          "color": "#1aeaf6ec",
                          "type": "solid",
                        },
                      ],
                      "border": Object {
                        "corner_radius": 28,
                      },
                      "height": Object {
                        "type": "match_parent",
                        "weight": 1,
                      },
                      "id": "main_screen_card_1_1",
                      "items": Array [
                        Object {
                          "items": Array [
                            Object {
                              "font_size": 32,
                              "font_weight": "medium",
                              "line_height": 40,
                              "text": "Москва",
                              "text_color": "#80ffffff",
                              "type": "text",
                            },
                          ],
                          "orientation": "horizontal",
                          "type": "container",
                        },
                        Object {
                          "delimiter_style": Object {
                            "color": "#0fff",
                          },
                          "height": Object {
                            "type": "match_parent",
                          },
                          "type": "separator",
                        },
                        Object {
                          "content_alignment_horizontal": "left",
                          "content_alignment_vertical": "center",
                          "items": Array [
                            Object {
                              "font_size": 90,
                              "font_weight": "medium",
                              "line_height": 90,
                              "text": "–3°",
                              "text_color": "#fff",
                              "type": "text",
                              "width": Object {
                                "type": "wrap_content",
                              },
                            },
                            Object {
                              "height": Object {
                                "type": "fixed",
                                "value": 100,
                              },
                              "image_url": "1",
                              "margins": Object {
                                "left": 10,
                              },
                              "type": "image",
                              "width": Object {
                                "type": "fixed",
                                "value": 100,
                              },
                            },
                          ],
                          "orientation": "horizontal",
                          "type": "container",
                        },
                        Object {
                          "font_size": 28,
                          "font_weight": "medium",
                          "line_height": 36,
                          "text": "Пасмурно.
Ощущается как -5°",
                          "text_color": "#80ffffff",
                          "type": "text",
                        },
                      ],
                      "paddings": Object {
                        "bottom": 24,
                        "left": 24,
                        "right": 24,
                        "top": 24,
                      },
                      "type": "container",
                      "width": Object {
                        "type": "match_parent",
                        "weight": 1,
                      },
                    },
                  ],
                  "margins": Object {
                    "top": 28,
                  },
                  "type": "container",
                  "width": Object {
                    "type": "match_parent",
                  },
                },
              ],
              "margins": Object {
                "left": 28,
              },
              "type": "container",
              "width": Object {
                "type": "match_parent",
              },
            },
            Object {
              "height": Object {
                "type": "match_parent",
              },
              "items": Array [
                Object {
                  "height": Object {
                    "type": "match_parent",
                  },
                  "items": Array [
                    Object {
                      "actions": Array [],
                      "background": Array [
                        Object {
                          "color": "#1aeaf6ec",
                          "type": "solid",
                        },
                      ],
                      "border": Object {
                        "corner_radius": 28,
                      },
                      "height": Object {
                        "type": "match_parent",
                        "weight": 1,
                      },
                      "id": "main_screen_card_0_2",
                      "items": Array [
                        Object {
                          "font_size": 32,
                          "font_weight": "medium",
                          "line_height": 40,
                          "text": "Youtube",
                          "text_color": "#80ffffff",
                          "type": "text",
                        },
                        Object {
                          "delimiter_style": Object {
                            "color": "#0fff",
                          },
                          "height": Object {
                            "type": "match_parent",
                            "weight": 31,
                          },
                          "type": "separator",
                        },
                        Object {
                          "alignment_horizontal": "center",
                          "alignment_vertical": "center",
                          "height": Object {
                            "type": "fixed",
                            "value": 112,
                          },
                          "image_url": "https://dialogs.s3.yandex.net/smart_displays/main-screen/cards/youtube/logo.png",
                          "margins": Object {
                            "top": 31,
                          },
                          "type": "image",
                          "width": Object {
                            "type": "fixed",
                            "value": 161,
                          },
                        },
                        Object {
                          "delimiter_style": Object {
                            "color": "#0fff",
                          },
                          "height": Object {
                            "type": "match_parent",
                            "weight": 56,
                          },
                          "type": "separator",
                        },
                      ],
                      "paddings": Object {
                        "bottom": 24,
                        "left": 24,
                        "right": 24,
                        "top": 24,
                      },
                      "type": "container",
                      "width": Object {
                        "type": "match_parent",
                        "weight": 1,
                      },
                    },
                  ],
                  "margins": Object {
                    "top": 0,
                  },
                  "type": "container",
                  "width": Object {
                    "type": "match_parent",
                  },
                },
                Object {
                  "height": Object {
                    "type": "match_parent",
                  },
                  "items": Array [
                    Object {
                      "actions": Array [],
                      "background": Array [
                        Object {
                          "color": "#1ae6eef9",
                          "type": "solid",
                        },
                        Object {
                          "content_alignment_horizontal": "center",
                          "content_alignment_vertical": "center",
                          "image_url": "https://yastatic.net/s3/dialogs/smart_displays/news/index.png",
                          "preload_required": 1,
                          "scale": "fill",
                          "type": "image",
                        },
                        Object {
                          "color": "#7a121316",
                          "type": "solid",
                        },
                      ],
                      "border": Object {
                        "corner_radius": 28,
                      },
                      "height": Object {
                        "type": "match_parent",
                        "weight": 1,
                      },
                      "id": "main_screen_card_1_2",
                      "items": Array [
                        Object {
                          "items": Array [
                            Object {
                              "font_size": 32,
                              "font_weight": "medium",
                              "line_height": 40,
                              "text": "Медуза",
                              "text_color": "#80ffffff",
                              "type": "text",
                            },
                          ],
                          "orientation": "horizontal",
                          "type": "container",
                        },
                        Object {
                          "delimiter_style": Object {
                            "color": "#0fff",
                          },
                          "height": Object {
                            "type": "match_parent",
                          },
                          "type": "separator",
                        },
                        Object {
                          "font_size": 28,
                          "font_weight": "medium",
                          "line_height": 36,
                          "max_lines": 5,
                          "text": "Tesla начала устанавливать терминалы Starlink на зарядных станциях Supercharger",
                          "text_color": "#e6ffffff",
                          "type": "text",
                        },
                      ],
                      "paddings": Object {
                        "bottom": 24,
                        "left": 24,
                        "right": 24,
                        "top": 24,
                      },
                      "type": "container",
                      "width": Object {
                        "type": "match_parent",
                        "weight": 1,
                      },
                    },
                  ],
                  "margins": Object {
                    "top": 28,
                  },
                  "type": "container",
                  "width": Object {
                    "type": "match_parent",
                  },
                },
              ],
              "margins": Object {
                "left": 28,
              },
              "type": "container",
              "width": Object {
                "type": "match_parent",
              },
            },
          ],
          "orientation": "horizontal",
          "paddings": Object {
            "left": 48,
            "right": 48,
          },
          "type": "container",
          "width": Object {
            "type": "match_parent",
          },
        },
        "state_id": 0,
      },
    ],
  },
  "templates": Object {},
}
`;

exports[`Main screen check snapshot without columns 1`] = `
Object {
  "card": Object {
    "log_id": "mainScreen",
    "states": Array [
      Object {
        "div": Object {
          "background": Array [
            Object {
              "color": "#121316",
              "type": "solid",
            },
          ],
          "height": Object {
            "type": "match_parent",
          },
          "items": Array [],
          "orientation": "horizontal",
          "paddings": Object {
            "left": 48,
            "right": 48,
          },
          "type": "container",
          "width": Object {
            "type": "match_parent",
          },
        },
        "state_id": 0,
      },
    ],
  },
  "templates": Object {},
}
`;

exports[`Main screen check snapshot without rows 1`] = `
Object {
  "card": Object {
    "log_id": "mainScreen",
    "states": Array [
      Object {
        "div": Object {
          "background": Array [
            Object {
              "color": "#121316",
              "type": "solid",
            },
          ],
          "height": Object {
            "type": "match_parent",
          },
          "items": Array [
            Object {
              "height": Object {
                "type": "match_parent",
              },
              "items": Array [],
              "margins": Object {
                "left": 0,
              },
              "type": "container",
              "width": Object {
                "type": "match_parent",
              },
            },
            Object {
              "height": Object {
                "type": "match_parent",
              },
              "items": Array [],
              "margins": Object {
                "left": 28,
              },
              "type": "container",
              "width": Object {
                "type": "match_parent",
              },
            },
            Object {
              "height": Object {
                "type": "match_parent",
              },
              "items": Array [],
              "margins": Object {
                "left": 28,
              },
              "type": "container",
              "width": Object {
                "type": "match_parent",
              },
            },
          ],
          "orientation": "horizontal",
          "paddings": Object {
            "left": 48,
            "right": 48,
          },
          "type": "container",
          "width": Object {
            "type": "match_parent",
          },
        },
        "state_id": 0,
      },
    ],
  },
  "templates": Object {},
}
`;
